unitTests/test_futils.cpp

```diff
@@ -105,4 +105,33 @@ TEST(urldecode, decodesGivenUrlInPlace)
     std::string url ("http%3a%2f%2fwww.geekhideout.com%2furlcode.shtml");
     urldecode(url);
     ASSERT_STREQ(expectedDecodedUrl.c_str(), url.c_str());
-}
\ No newline at end of file
+}
+
+TEST(base64encode, encodesValidString)
+{
+    const std::string original ("This is a unit test");
+    const std::string expected ("VGhpcyBpcyBhIHVuaXQgdGVzdA==");
+    size_t encodeLength = ((original.size() + 2) / 3) * 4 + 1;
+    char * result = new char [encodeLength];
+    ASSERT_EQ(1, base64encode(original.c_str(), original.size(), result, encodeLength));
+    ASSERT_STREQ(expected.c_str(), result);
+    delete [] result;
+}
+
+TEST(base64encode, doesNotEncodeWithNotBigEnoughResultSize)
+{
+    const std::string original ("This is a unit test");
+    size_t encodeLength = (original.size());
+    char * result = new char [encodeLength];
```

Comment:
Wait, this thing doesn't get `delete`'d, why does ASAN not complain?

Reply:
I actually discovered the issue thanks to ASAN :wink: